tools/validators/instance_validator/instance_parser.py

```diff
@@ -100,26 +100,36 @@ def parse_yaml(filename):
     filename: filepath location of the YAML file
 
   Returns:
-    Returns the parsed YAML data in a stricyaml-provided
-    datastructure which is similar to a Python dictionary.
+    Returns a tuple with the parsed YAML data in a stricyaml-provided
+    datastructure which is similar to a Python dictionary, and a
+    possible exception.
   """
-  yaml = _load_yaml_with_schema(filename, _SCHEMA)
+  yaml, err = _load_yaml_with_schema(filename, _SCHEMA)
 
   if yaml is None:
-    return None
+    return (None, err)
```

Comment:
returning the exception is enough why the need to return None as well?

Reply:
Because the function is now defined to return a tuple, so this avoids the errors when unpacking the return values later

Reply:
I think the correct thing to do here would be to catch and extract the relevant info from all the possible, then repackage it into a single container that callers can handle.  I would then decide whether this container is an exception to be THROWN or a non-exception container that gets passed as an element in the tuple and the caller needs to know to check based on whether the application can reasonably can continue through these errors.  if not.  the correct thing to do is simply throw the exception.

Reply:
Okay, I've refactored with the heuristic of "throw the exception if the application cannot reasonably continue through the error." This means I overhauled the YAML schema syntax parsing, because any error at that step is unreasonable to continue from. To make sure the error messages are clean, I'm printing the exception messages and using `sys.exit(0)`.